EXT_BeamPhysics.md

```diff
@@ -0,0 +1,269 @@
+
+- `referenceEnergy`
+    - Type: Optional *(float)* attribute.
+    - Description: Specifies the reference energy from which the `energy` is measured with respect to.
+
+- `referenceTotalMomentum`
+    - Type: Optional *(float)* attribute
+    - Description: Specifies the reference total momentum from which the total momentum is measured with respect to.
+
+- `totalCharge`
+    - Type: Optional *(float)* attribute.
+    - Description: The total charge of all the particles.
+
+Per-Particle Records of the `Particle Root Group`
+---------------------
+
+The following records store data on a particle by particle basis.
+
+- `time-refTime`
+- Type: Optional *(float)*
+- Description: Particle time minus the referece time.
+
+- `energy/`
+  - Type: Optional *(float)*
+  - Description: The total energy of the particles relative to the `referenceEnergy` attribute.
+
+- `electricField/`
+    - Type: Optional 2-component *(float)*
+    - Description: Electric field. Used for photons only.
+    - Components: (`x`, `y`).
+        - For each component, the field is specified using either (`amplitude`, `phase`) or (`Real`, `Imaginary`)
+        subcomponents.
+
+- `macroCharge/`
+    - Type: Optional *(float)*
+    - Description: "Macro"-particles are simulation particles that represent multiple real particles.
+    The `macroCharge` is the the total charge of the collection of particles that a macro-particle
+    represents.
+
+- `momentum/`
+    - Type: Optional 2 or 3-vector *(float)*
+    - Description: The total momentum of the particles relative to the `momentumOrigin` attribute.
+    - Components: (`px`, `py`) or (`px`, `py`, `pz`).
+    - Note: A program using phase space coordinates to describe particle positions will typically use the transverse momentum
+    `px` and `py` along with the `totalMomentum` or `energy` records.
+
+- `pathLength/`
+    - Type: Optional *(float)*
+    - Description: Length that a particle has traveled.
+
+- `position`
+    - Type: Required 3-vector *(float)*
+    - Description: Position relative to the coordinate origin.
+    - Components: (`x`, `y`, `<z-coord>`) where `<z-coord>` is one of:
+        - `z`: True longitudinal coordinate.
+        - `-beta.c.dt`: Phase space coordinate conjugate to a momentum based "pz".
+        - `-beta.c.t`:Phase space coordinate conjugate to a momentum based "pz".
+        - `-c.dt`: Phase space coordinate conjugate to an energy based "pz".
+        - `-c.t`: Phase space coordinate conjugate to an energy based "pz".
+        - Where: **beta** = particle speed, **c** = speed of light, **t** = time, and **dt** = time relative to the reference time.
+
+- `refTime/`
+    - Type: Optional *(float)*
+    - Description: The reference particle time.
+    Note that the reference time is a function of **s** and therefore can be different for different particles.
+
+- `s-Position`
+    - Type: Optional *(float)*
+    - Description: Absolute longitudinal distance in **lattice** coordinates.  
```

Comment:
Could you explain the difference with the `z` position above?

Reply:
**s**-position is the distance along the **s**-axis of the lattice coordinate system. 

**z** is the 5th component of the particle's phase space position. The definition of **z** is different for different programs. For example, with my Bmad software, **z** is defined to be -beta*c*(t_ref-t) where beta is the particle speed, c is the speed of light, t_ref is the time associated with the reference particle, and t is the time associated with the particle under consideration. The reason for the funny definition is to have canonical coordinates. 

One thing to keep in mind is that there is no simple relationship between **s** and **z**.